File: TypeScript/Codewars/src/resolutions/Maximum-Product.ts
export function adjacentElementsProduct(arr: number[]): number {
	let result = Number.MIN_SAFE_INTEGER;

	for (let index = 0; index < arr.length - 1; index++) {
		result = Math.max(result, arr[index] * arr[index + 1]);
	}

	return result;
}

console.log(adjacentElementsProduct([-23, 4, -5, 99, -27, 329, -2, 7, -921]));

File: Csharp/CodeWars/src/Completed/MaximumProduct.cs
namespace CodeWars.Completed;

public class MaximumProduct
{
    public static int AdjacentElementsProduct(int[] array)
    {
        int result = int.MinValue;

        for (int index = 0; index < array.Length - 1; index++)
            result = Math.Max(result, array[index] * array[index + 1]);

        return result;
    }
}
